zerver/lib/actions.py

```diff
@@ -2582,8 +2583,11 @@ def bulk_get_subscriber_user_ids(stream_dicts: Iterable[Mapping[str, Any]],
         stream_recipient.populate_with(stream_id=stream_dict["id"],
                                        recipient_id=stream_dict["recipient_id"])
         try:
-            validate_user_access_to_subscribers_helper(user_profile, stream_dict,
-                                                       lambda: sub_dict[stream_dict["id"]])
+            validate_user_access_to_subscribers_helper(
+                user_profile,
+                stream_dict,
+                lambda user_profile: sub_dict[stream_dict["id"]],
```

Comment:
I hate that this interface accepts a `UserProfile` object and totally ignores it.  Not that I liked the old interface.  This is probably at least worth a comment here that we're doing something weird?

(The whole thing is a mess because we're trying to defer fetching the UserProfile object in the other code path).

Reply:
I figured if you wanted to see why `validate_user_access_to_subscribers_helper` passes us a `UserProfile`, you’d look at `validate_user_access_to_subscribers_helper`, which has an explanatory docstring.